ansible/library/interface_facts.py

```diff
@@ -196,8 +184,8 @@ def main():
             promisc_mode = (data & 0x0100 > 0)
             interfaces[device]['promisc'] = promisc_mode
 
-        def parse_ip_output(module, output, secondary=False):
-            for line in output.split('\n'):
+        def parse_ip_output(output, secondary=False):
+            for line in output.split('\\n'):
```

Comment:
Is this because you didn't call run_command()?

Reply:
This same entire logic of retrieving and parsing the interface details will now be part of the python script which the ansible module creates. ( INTF_IP_GET_INFO_SCRIPT = "/tmp/gather_intf_ip_info.py"). Hence removed it from this function as it will be called in the script.